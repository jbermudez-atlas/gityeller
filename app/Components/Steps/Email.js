import React from 'react';
import isEmail from 'validator/lib/isEmail';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ''
    };
  }
  stepLoaded() {
    return this.props.activeStep === 3;
  }
  next() {
    this.props.next({
      email: this.state.email,
      activeStep: this.props.activeStep + 1
    });
  }
  validateEmail(e) {
    if (isEmail(e.target.value)) {
      this.setState({
        email: e.target.value
      });
    }

    if (e.keyCode === 13) {
      this.next();
    }
  }
  render() {
    return (
      <div className={this.stepLoaded() ? this.props.styles.activeStep : this.props.styles.inactiveStep}>
        <div className="form-group">
          <label htmlFor="email">Email</label>
          <input
            className={`form-control ${this.props.styles.customControl}`}
            id="email"
            type="text"
            onKeyUp={this.validateEmail.bind(this)}
            placeholder="konmpar@gmail.com"
            autoComplete="off"
          />

          <div className={`row ${this.props.styles.actions}`}>
            <div className="col-xs-6">
              <button
                className={`btn btn-default ${this.props.styles.customDefaultButton}`}
                onClick={this.props.startOver}>
                  Reset
              </button>
            </div>
            <div className="col-xs-6 text-right">
              {this.state.email &&
                <button
                  className={`btn btn-default ${this.props.styles.customPrimaryButton}`}
                  disabled={!this.state.email}
                  onClick={this.next.bind(this)}>
                    Next
                </button>
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}
