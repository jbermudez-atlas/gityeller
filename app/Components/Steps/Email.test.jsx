import React from 'react';
import Email from './Email';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

test('Take Snapshot', () => {
  const component = renderer.create(
    <Email
      activeStep={3}
      next={function(){}}
      startOver={function(){}}
      styles={{}}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Will show Next button on valid email', () => {
  const checkbox = shallow(
    <Email
      activeStep={3}
      next={function(){}}
      startOver={function(){}}
      styles={{}}/>
  );

  checkbox.find('input').simulate('keyup', {target: {value: 'konmpar@gmail.com'}});

  expect(checkbox.find('.btn').length).toBe(2);
});

it('Will not show Next button on not valid email', () => {
  const checkbox = shallow(
    <Email
      activeStep={3}
      next={function(){}}
      startOver={function(){}}
      styles={{}}/>
  );

  checkbox.find('input').simulate('keyup', {target: {value: 'not.a.valid.email'}});

  expect(checkbox.find('.btn').length).toBe(1);
});

it('Will call next function on Enter with a valid email', () => {
  const spy = jest.fn();
  const checkbox = shallow(
    <Email
      activeStep={3}
      next={spy}
      startOver={function(){}}
      styles={{}}/>
  );

  checkbox.find('input').simulate('keyup', {keyCode: 13, target: {value: 'konmpar@gmail.com'}});

  expect(spy).toBeCalled();
});

it('Will not call next function on Enter with a not valid email', () => {
  const spy = jest.fn();
  const checkbox = shallow(
    <Email
      activeStep={3}
      next={spy}
      startOver={function(){}}
      styles={{}}/>
  );

  checkbox.find('input').simulate('keyup', {keyCode: 13, target: {value: 'not.a.valid.email'}});

  expect(spy.mock.calls.length).toBe(0);
});
