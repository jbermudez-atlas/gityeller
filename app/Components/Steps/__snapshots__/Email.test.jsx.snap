exports[`test Take Snapshot 1`] = `
<div
  className={undefined}>
  <div
    className="form-group">
    <label
      htmlFor="email">
      Email
    </label>
    <input
      autoComplete="off"
      className="form-control undefined"
      id="email"
      onKeyUp={[Function]}
      placeholder="konmpar@gmail.com"
      type="text" />
    <div
      className="row undefined">
      <div
        className="col-xs-6">
        <button
          className="btn btn-default undefined"
          onClick={[Function]}>
          Reset
        </button>
      </div>
      <div
        className="col-xs-6 text-right">
        
      </div>
    </div>
  </div>
</div>
`;
